spec.html

```diff
@@ -34849,17 +34848,20 @@ <h1>
           </emu-note>
         </emu-clause>
 
-        <emu-clause id="sec-tozeropaddeddecimalstring" type="abstract operation">
+        <emu-clause id="sec-zeropad" type="abstract operation" oldids="sec-tozeropaddeddecimalstring">
           <h1>
-            ToZeroPaddedDecimalString (
+            ZeroPad (
               _n_: a non-negative integer,
               _minLength_: a non-negative integer,
+              _alphabet_: ~decimal~, ~hex-lower~, or ~hex-upper~,
             ): a String
           </h1>
           <dl class="header">
           </dl>
           <emu-alg>
-            1. Let _S_ be the String representation of _n_, formatted as a decimal number.
+            1. If _alphabet_ is ~hex-lower~, let _S_ be the String representation of _n_, formatted as a lowercase hexadecimal number.
+            1. Else if _alphabet_ is ~hex-upper~, let _S_ be the String representation of _n_, formatted as an uppercase hexadecimal number.
+            1. Else, let _S_ be the String representation of _n_, formatted as a decimal number.
```

Comment:
This could also use the approach of [proposal-regex-escaping](https://tc39.es/proposal-regex-escaping/#sec-encode) and replace prose with AO calls, e.g.
```suggestion
            1. If _alphabet_ is ~decimal~, let _S_ be BigInt::toString(ℤ(_n_), 10).
            1. Else if _alphabet_ is ~hex-lower~, let _S_ be BigInt::toString(ℤ(_n_), 16).
            1. Else, let _S_ be the ASCII-uppercase of BigInt::toString(ℤ(_n_), 16).
```
(importing [ECMA-402's ASCII-uppercase](https://tc39.es/ecma402/#sec-case-sensitivity-and-case-mapping), as is already planned for [Temporal](https://tc39.es/proposal-temporal/#sec-ecmascript-language-types-string-type))